vscode/src/createProject.ts

```diff
@@ -265,93 +254,175 @@ export async function initProjectCreator(context: vscode.ExtensionContext) {
-        const projectChoice = await vscode.window.showQuickPick(
-          projectChoices.map((choice) => {
-            if ("github" in choice.ref) {
-              return {
-                label: choice.name,
-                detail: `github://${choice.ref.github.owner}/${choice.ref.github.repo}#${choice.ref.github.ref}`,
-                iconPath: githubIcon,
-                ref: choice.ref,
-              };
-            } else {
-              return {
-                label: choice.name,
-                detail: choice.ref.path,
-                iconPath: folderIcon,
-                ref: choice.ref,
-              };
-            }
-          }),
-          { placeHolder: "Pick a project to add as a reference" },
-        );
-
-        if (!projectChoice) {
-          log.info("User cancelled project choice");
+        if (!importChoice) {
+          log.info("User cancelled import choice");
           return;
         }
 
-        log.info("User picked project: ", projectChoice);
-
-        if (!manifestObj["dependencies"]) manifestObj["dependencies"] = {};
-        manifestObj["dependencies"][projectChoice.label] = projectChoice.ref;
-
-        // Apply the edits to the qsharp.json
-        const edit = new vscode.WorkspaceEdit();
-        edit.replace(
-          qsharpJsonUri,
-          new vscode.Range(0, 0, qsharpJsonDoc.lineCount, 0),
-          JSON.stringify(manifestObj, null, 2),
-        );
-        if (!(await vscode.workspace.applyEdit(edit))) {
-          vscode.window.showErrorMessage(
-            "Unable to update the qsharp.json file. Check the file is writable",
+        if (importChoice === "Import from GitHub") {
+          await importExternalPackage(
+            qsharpJsonDoc,
+            qsharpJsonUri,
+            manifestObj,
+          );
+        } else {
+          await importLocalPackage(
+            qsharpJsonDoc,
+            qsharpJsonDir,
+            qsharpJsonUri,
+            manifestObj,
           );
-          return;
         }
-
-        // Bring the qsharp.json to the front for the user to save
-        await vscode.window.showTextDocument(qsharpJsonDoc);
       },
     ),
   );
+  async function importLocalPackage(
+    qsharpJsonDoc: vscode.TextDocument,
+    qsharpJsonDir: vscode.Uri,
+    qsharpJsonUri: vscode.Uri,
+    manifestObj: any,
+  ) {
+    // Find all the other Q# projects in the workspace
+    const projectFiles = (
+      await vscode.workspace.findFiles("**/qsharp.json")
+    ).filter((file) => file.toString() !== qsharpJsonUri.toString());
+
+    const projectChoices: Array<{ name: string; ref: LocalProjectRef }> = [];
+
+    projectFiles.forEach((file) => {
+      const dirName = file.path.slice(0, -"/qsharp.json".length);
+      const relPath = getRelativeDirPath(qsharpJsonDir.path, dirName);
+      projectChoices.push({
+        name: dirName.slice(dirName.lastIndexOf("/") + 1),
+        ref: {
+          path: relPath,
+        },
+      });
+    });
```

Comment:
Would look more natural to write this as a `.map()` in my opinion.

Reply:
again, this is just relocated code, but i've rewritten it as a map